Add UnexpectedCancelledError

kyleknap opened the pull request:
Introduced a new type of ``CancelledError`` called ``UnexpectedCancelledError`` that differentiates between a cancellation by the user or cntrl-c and a general error hit in the context manger of the TransferManager.

cc @jamesls @JordonPhillips

jamesls commented on s3transfer/futures.py:
```diff
@@ -221,16 +222,23 @@ def result(self):
             return self._result
 
-    def cancel(self, msg=''):
+    def cancel(self, msg='', by_user=True):
```
Rather than `by_user`, what about providing your own cancelled error type.  That way we can extend this in the future for new cancellation types without having to add more params.

kyleknap replied:
Makes sense

jamesls commented on s3transfer/manager.py:
```diff
@@ -482,14 +482,19 @@ def __enter__(self):
-        self.shutdown(cancel, cancel_msg)
+            # If it was a KeyboardInterrupt, the cancellation was initiated
+            # by the user.
+            if exc_type == KeyboardInterrupt:
```
`isinstance`

kyleknap commented:
@jamesls I updated based on feedback. I also changed ``UnexpectedCancelledError`` to ``FatalError`` because I did not like that name very much. Should be good to look at again.
